subsys/net/ip/l2/wifi_mgmt.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_NET_DEBUG_L2_WIFI_MGMT)
+#define SYS_LOG_DOMAIN "net/wifi_mgmt"
+#define NET_LOG_ENABLED 1
+#endif
+
+#include <errno.h>
+
+#include <net/net_core.h>
+#include <net/net_if.h>
+#include <net/wifi_mgmt.h>
+
+static int wifi_connect(u32_t mgmt_request, struct net_if *iface,
+			void *data, size_t len)
+{
+	struct wifi_connect_req_params *params =
+		(struct wifi_connect_req_params *)data;
+
+	if ((params->security > WIFI_SECURITY_TYPE_PSK) ||
+	    (params->psk_length < 8) || (params->psk_length > 64) ||
+	    (params->ssid_length > WIFI_SSID_MAX_LEN) ||
+	    ((params->security == WIFI_SECURITY_TYPE_PSK) &&
+	     (params->psk_length == 0)) ||
+	    ((params->channel != WIFI_MGMT_CHANNEL_ANY) &&
+	     (params->channel > WIFI_MGMT_CHANNEL_MAX)) ||
+	    !params->ssid || !params->psk) {
+		return -EINVAL;
+	}
+
+	if (net_if_is_ip_offloaded(iface)) {
+		struct net_wifi_mgmt_offload *off_api =
+			(struct net_wifi_mgmt_offload *) iface->dev->driver_api;
+
+		return off_api->connect(iface->dev, params);
+	}
+
+	return -ENETDOWN;
+}
+
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_WIFI_CONNECT, wifi_connect);
+
+static void _scan_result_cb(struct net_if *iface,
+			    struct wifi_scan_result *entry)
+{
+	if (!iface || !entry) {
+		return;
+	}
+
+	net_mgmt_event_notify_with_info(NET_EVENT_WIFI_SCAN_RESULT, iface,
```

Comment:
@@In the case of SimpleLink wifi scan, the table of scan results will be available from the chip all at once.   So, during the wifi_scan() call, the _scan_result_cb() can be called up to 30 times (max) in rapid succession.

Now, the net_mgmt server thread uses an event queue to store notifications for later delivery.

Depending on the values of 
`CONFIG_NET_MGMT_EVENT_QUEUE_SIZE (default 2)`,
and `CONFIG_NET_MGMT_EVENT_THREAD_PRIO (default 7)`,
and the priority of the thread calling wifi_scan(), 
it seems the server thread could drop notifications.

One solution would be to force `CONFIG_NET_MGMT_EVENT_QUEUE_SIZE` to be the maximum table size (30) in this case; but this will be wasting extra memory - especially for an operation that is infrequently executed.

So, @jukkar 's suggestion of passing in the scan_results table would nicely avoid that issue here...